### PR TITLE
Split up cmake into different targets

### PR DESCRIPTION
**Pull request by nidegen**
2 or 3 things to fix (x86 flags, iOS linking), but already quite mergeable for me. 

depending on how much merge conflict potential you see we can also wait..

### CODE REVIEW

**Comment by nidegen** [comment]
OK, now everything should be fixed:)

File: include/openmapper/input_source.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ class InputSource {
                   // ROS, native mobile framework...
   };
   InputSource();
-  virtual ~InputSource();
+  // virtual ~InputSource();
```
Comment (nidegen) [inline include/openmapper/input_source.h:31]: I had to comment out those virtual destructors (see below too) in order to build it for iOS. do those currently have any crucial function?

Comment (gocarlos) [reply include/openmapper/input_source.h:31]: no, these were created automatically by eclipse...

**Comment by gocarlos** [comment]
I could not test the ROS build, but you can merge, I'll try this later today :)

File: wrapper/ros/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+################################
+# OpenMapper
+################################
+
+PROJECT("OpenMapperROS")
+
+SET(OPENMAPPER_ROS_NODE "OpenMapperROS")
+
+# Request C++11 standard from compiler.
+SET(CMAKE_CXX_STANDARD 11)
```
Comment (gocarlos) [inline wrapper/ros/CMakeLists.txt:12]: I think such code should be in the main Cmakelits

**Review by gocarlos — APPROVED** [review]

File: wrapper/ros/CMakeLists.txt (hunk 2 above)
Comment (nidegen) [reply wrapper/ros/CMakeLists.txt:12]: sure, that should be moved. I merge anyways and fix it afterwards

**Pull request merged**